### PR TITLE
Add Mixed Precision (AMP) Support to Predictive Coding Training

### PR DESCRIPTION
**Pull request by YonatanBest**
- Wrapped all major tensor operations in step_embed, step_linear, and step_attn (in utils/pc_utils.py) with torch.amp.autocast('cuda') when running on CUDA.
- Ensured that all predictive coding local update steps benefit from mixed precision, without altering the custom local learning logic.
- Updated the autocast API to the latest PyTorch standard to avoid deprecation warnings.
- No Changes to Training Logic

without the amp
real    77m1.387s
user    63m26.574s
sys     13m11.352s

With the amp added
real    75m15.791s
user    60m1.588s
sys     14m52.895s

- Overall training time reduced by ~2%.
- No observed loss in model stability or accuracy.

### CODE REVIEW

**Comment by Saad-Musema** [comment]
Could you send a different PR for the AMP changes? I want you to send the PR to the Optimization-Clean branch. I also want you to make the "If not requires_update" sections out of the "with autocast" argument, since we do not want the weights to be with F16.
